lib/asset_cloud/asset.rb

```diff
@@ -34,7 +34,13 @@ def self.at(cloud, key, value = nil, metadata = nil, &block)
     end
 
     def <=>(other)
-      cloud.object_id <=> other.cloud.object_id && key <=> other.key
+      return 1 unless other.is_a?(Asset)
+      compare_cloud = cloud.object_id <=> other.cloud.object_id
```

Comment:
Not sure if I understand why we use the `<=>` operator here? Shouldn't it just be `cloud.object_id == other.cloud.object_id && key == other.key`

Ruby `==` operator will call `<=>` if needed but I don't think we need to do it ourselves

Reply:
The `<=>` can't return `true/false` it needs to return 0, 1 and -1. If we use `==` we will return `true/false`.

Reply:
Ah yeah that's true